Sources/_MatchingEngine/Regex/AST/ASTBuilder.swift

```diff
@@ -0,0 +1,156 @@
+/*
+
+These functions are temporary AST construction helpers. As
+the AST gets more and more source-location tracking, we'll
+want easier migration paths for our parser tests (which
+construct and compare location-less AST nodes) as well as the
+result builder DSL (which has a different notion of location).
+
+Without real namespaces and `using`, attempts at
+pseudo-namespaces tie the use site to being nested inside a
+type. So for now, these are global, but they will likely be
+namespaced in the future if/when clients are weaned off the
+AST.
+
+*/
+
```

Comment:
For namespacing, could you move these top-level symbols to an enum, e.g. `ASTBuilder`, as static methods?

Reply:
Maybe eventually. I'm inclined not to for now because otherwise you end up either with a bunch of `ASTBuilder.concat(ASTBuilder.atom....)` in unit tests, or every client just reinvents these. Since the contextual type requested is AST and not ASTBuilder, they wouldn't be seen for lookup unless everything was typed as a builder instead of an AST. These don't work well hosted on AST because names can collide with case names.

Longer term, I don't think we'll be as enthusiastically creating AST nodes for every need. This is public in the "core" module right now (so visible to _StringProcessing but not its clients). Another option is to make these internal inside _StringProcessing and the test use case can use `@testable` anyways.

Also, we'll see if AST even stays as an enum or in its current form. We might need to overhaul it more for options tracking.

Reply:
>  Since the contextual type requested is AST and not ASTBuilder, they wouldn't be seen for lookup unless everything was typed as a builder instead of an AST. 

Hmm, these could also be static methods on `AST` itself though. I'm still concerned about polluting the global namespace.

> Another option is to make these internal inside _StringProcessing and the test use case can use `@testable` anyways.

I think that this approach works better; we only build location-less ASTs in  _StringProcessing module and its tests.